packages/lib/zuauth/src/server.ts

```diff
@@ -52,17 +55,27 @@ export async function authenticate(
       isEqualEdDSAPublicKey(pubKey, pcd.claim.signer)
     )
   ) {
-    throw new Error(
+    throw new ZuAuthAuthenticationError(
       "Signing key does not match any of the configured public keys"
     );
   }
 
+  if (
+    eventIds.size > 0 &&
+    pcd.claim.partialTicket.eventId &&
+    !eventIds.has(pcd.claim.partialTicket.eventId)
+  ) {
+    throw new ZuAuthAuthenticationError(
+      "Event ID does not match any of the configured event IDs"
+    );
+  }
+
   if (
     productIds.size > 0 &&
     pcd.claim.partialTicket.productId &&
```

Comment:
As with eventID above, I think it might be necessary to throw here if the productID isn't revealed when it should be.